fe/fe-core/src/main/java/org/apache/doris/nereids/NereidsPlanner.java

```diff
@@ -286,6 +289,12 @@ private void analyze(boolean showPlanProcess) {
             LOG.debug("Start analyze plan");
         }
         keepOrShowPlanProcess(showPlanProcess, () -> cascadesContext.newAnalyzer().analyze());
+        if (this.cascadesContext.getConnectContext().getSessionVariable().isEnableDmlMaterializedViewRewrite()
```

Comment:
Don't distinguish between the inner and external table?

Reply:
if sql contains external table and the `set enable_dml_materialized_view_rewrite_when_base_table_unawareness = false;` can also allow external table.
this control logic is in
`Env.getCurrentEnv().getMtmvService().getRelationManager()
                .getAvailableMTMVs(usedBaseTables, cascadesContext.getConnectContext(),
                        true, ((connectContext, mtmv) -> {
                            return MTMVUtil.mtmvContainsExternalTable(mtmv) && (!connectContext.getSessionVariable()`